python/cog/types.py

```diff
@@ -218,13 +218,27 @@ def __init__(self, url: str, filename: Optional[str] = None) -> None:
         object.__setattr__(self, "name", filename)
         object.__setattr__(self, "__url__", url)
 
+    def __del__(self) -> None:
+        try:
+            object.__getattribute__(self, "__target__")
+        except AttributeError:
+            # Do nothing when tearing down the object if the response object
+            # hasn't been created yet.
+            return
+
+        super().__del__()
```

Comment:
This showed up when I removed the requests mocking from the tests. It turns out that by default instances of `io.BaseIO` do some teardown (I assume when under test) that then results in trying to access various attributes which in turn triggers a request.

This change prevents that teardown unless we've actually made a request.